File: LR_3-4/Pack3.pas
{ Задача 3.21 }

var
  res_data: array[1..100] of record
    cat: string;
    value: integer;
  end;
  i, sum_res, cnt: integer;
  cat: string;
  inp_val: integer;

begin
  cnt := 0;
  
  while True do
  begin
    WriteLn('Введите категорию или "стоп" для окончания подсчёта: ');
    ReadLn(cat);

    if cat = 'стоп' then
    begin
      sum_res := 0;
      
      for i := 1 to cnt do
      begin
        sum_res := sum_res + res_data[i].value;
        WriteLn('Категория ', res_data[i].cat, ': ', res_data[i].value);
      end;
      
      WriteLn('Общая сумма: ', sum_res);
      Break;
    end;
    
    cnt := cnt + 1;
    WriteLn('Введите данные по категории ', cat, ': ');
    ReadLn(inp_val);
    
    res_data[cnt].cat := cat;
    res_data[cnt].value := inp_val;
  end;
end.

File: LR/LR_3-4/Pack3.py
# Задача 3.21

res_data = dict()

while True:
    cat = str(input("Введите категорию или 'стоп' для окончания подсчёта: "))

    if(cat == "стоп"):
        sum_res = 0
        for cat in res_data:
            sum_res += res_data[cat]
            print(f"Категория '{cat}': {res_data[cat]}")
        
        print(f"Общая сумма: {sum_res}")
        break

    res_data[cat] = int(input(f"Введите данные по категории {cat}: "))